### PR TITLE
Refactor heart beat task, use json to serialize/deserialize

### PR DESCRIPTION
**Pull request by ruanwenjun**
## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

(or)

If your pull request contain incompatible change, you should also add it to `docs/docs/en/guide/upgrede/incompatible.md`

### CODE REVIEW

File: dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/model/BaseHeartBeatTask.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        }
+    }
+
+    public void shutdown() {
+        log.warn("{} task finished", threadName);
+        runningFlag = false;
+    }
+
+    private void handleInterruptException(InterruptedException ex) {
+        shutdown();
```
Comment (caishunfeng) [inline dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/model/BaseHeartBeatTask.java]: should not shutdown when interrupt.

File: dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/registry/ServerNodeManager.java [hunk 2]
```diff
@@ -379,19 +343,24 @@ private void syncMasterNodes(Collection<String> nodes, List<Server> masterNodes)
...
             workerGroupNodes.put(workerGroup, workerNodes);
         } finally {
-            workerGroupLock.unlock();
+            workerGroupWriteLock.unlock();
         }
     }
 
     public Map<String, Set<String>> getWorkerGroupNodes() {
-        return Collections.unmodifiableMap(workerGroupNodes);
+        workerGroupLock.readLock();
```
Comment (caishunfeng) [inline dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/registry/ServerNodeManager.java]:
```suggestion
        workerGroupReadLock.lock();
```

File: dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/registry/ServerNodeManager.java [hunk 3]
```diff
@@ -379,19 +343,24 @@ private void syncMasterNodes(Collection<String> nodes, List<Server> masterNodes)
...
+        try {
+            return Collections.unmodifiableMap(workerGroupNodes);
+        } finally {
+            workerGroupLock.writeLock();
```
Comment (caishunfeng) [inline dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/registry/ServerNodeManager.java]:
```suggestion
            workerGroupReadLock.unlock();
```

File: dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/model/BaseHeartBeatTask.java (hunk 1 above)
Comment (ruanwenjun) [reply dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/model/BaseHeartBeatTask.java]: Done.

**Review by caishunfeng — APPROVED** [review]
+1

**Pull request merged**